Library/Homebrew/cask/audit.rb

```diff
@@ -410,17 +410,48 @@ def audit_token_valid
       add_error "cask token underscores should be replaced by hyphens" if cask.token.include? "_"
       add_error "cask token should not contain double hyphens" if cask.token.include? "--"
 
-      if cask.token.match?(/[^@a-z0-9-]/)
-        add_error "cask token should only contain lowercase alphanumeric characters, hyphens and @"
-      end
-
       if cask.token.start_with?("-", "@") || cask.token.end_with?("-", "@")
         add_error "cask token should not have leading or trailing hyphens and/or @"
       end
 
-      add_error "cask token @ unrelated to versioning should be replaced by -at-" if cask.token.count("@") > 1
-      add_error "cask token should not contain a hyphen followed by @" if cask.token.include? "-@"
-      add_error "cask token should not contain @ followed by a hyphen" if cask.token.include? "@-"
+      unversioned_token, _, version_designation = cask.token.rpartition("@")
+      if unversioned_token.empty?
+        match_data = /-(?<designation>alpha|beta|rc|release-candidate)$/.match(cask.token)
+        if match_data && cask.tap&.official? && cask.tap != "homebrew/cask-versions"
```

Comment:
```suggestion
        if match_data && cask.tap&.official? && cask.tap.name != "homebrew/cask-versions"
```

Reply:
I was wondering if we should only apply to `homebrew/cask` given audit failures in fonts. From original location of check (which I'll probably move code back to later), I'm guessing we either don't apply that audit or there were no new fonts with `-beta|alpha`.

As I recall, we have automation there that picks those names?

Reply:
We should definitely fix the automation, but maybe this should only apply to `brew audit --new` for now?